### PR TITLE
Project overview usage of model form

### LINKED ISSUE #81: Create and update projects via `ModelForm`
- Switch the current implementation of the `CreateNewProjectFrom` and the `UpdateProjectForm` to use the ModelForm class instead
- Move all the form validation to the form itself to remove the current edge case where a project name must be unique error is raised, even if the project is unique but other data of the form is invalid

### PR DESCRIPTION
**Pull request by plhrtr**
>[!WARNING]
>- As mentioned [here](https://github.com/ARTIST-Association/CANVAS/settings/secrets/actions) `[Secrets] are not passed to workflows that are triggered by a pull request from a fork.`, so manual execution for the test is needed here. _(They run fine locally on my machine)_

This PR makes more use of the Django model form for both the project creation and updating process.

Also now the create project model shows the occurring error directly and not via the message system.

Closes #81

### CODE REVIEW

**Review by lisannengel — APPROVED** [review]
lgtm :)

File: canvas_editor/project_management/tests/test_views/test_project_page.py [hunk 1]
```diff
@@ -247,7 +246,7 @@ def test_update_project_post_name_description_changed(self):
         self.assertEqual(Project.objects.count(), 1)
 
     def test_update_project_post_name_description_changed_description_is_empty(self):
```
Comment (fluegelk) [inline canvas_editor/project_management/tests/test_views/test_project_page.py]: 1. Test name seems outdated (description is no longer empty)
2. Do we still need a test for an empty description (as the old test)?

**Review by fluegelk — CHANGES_REQUESTED** [review]
Nice work! Please pull the updates from main, that should allow to CI to run on PRs. There are 5 comments that should be quick to resolve. 
I feel like there are still some possibilities for improvement in the form/model interaction with where code is placed and reducing duplicate validation code, but in the interest of time I would leave that as is for now.

File: canvas_editor/project_management/forms/create_project_form.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    class Meta:
+        """Meta information for this model form."""
+
+        model = Project
+        fields = ["name", "description"]
+
+    def clean_name(self):
+        """Check whether the name contains any special characters."""
+        project_name = str(self.cleaned_data.get("name")).strip().replace(" ", "_")
```
[the remark this answers is not in the document]
Comment (plhrtr) [reply canvas_editor/project_management/forms/create_project_form.py]: Spaces break the Js selectors e.g. opening the correct bootstrap modal. 
And yes, currently other white space characters seem to allowed, but they also break the selectors. 

We could store the names in the database normally, and replace all whitespace characters with hyphens or underscores, but this would lead to some really weird issues with unique names. 
Or I could also replace all other white space characters with underscores.

File: canvas_editor/project_management/tests/test_views/test_project_page.py (hunk 1 above)
Comment (plhrtr) [reply canvas_editor/project_management/tests/test_views/test_project_page.py]: I think we could remove the test, as it wouldn't really test anything new that the test with an updated description does

**Review by fluegelk — APPROVED** [review]
